src/main/resources/db/migration/v0.32.0.1__gene_molecular_interaction.sql

```diff
@@ -0,0 +1,104 @@
+	interactionid varchar (255),
+	uniqueid varchar (2000),
+	interactionsource_id bigint,
+	interactiontype_id bigint,
+	interactorarole_id bigint,
+	interactorbrole_id bigint,
+	interactoratype_id bigint,
+	interactorbtype_id bigint
+);
+
+ALTER TABLE geneinteraction ADD CONSTRAINT geneinteraction_id_fk FOREIGN KEY (id) REFERENCES genegeneassociation (id);
+ALTER TABLE geneinteraction ADD CONSTRAINT geneinteraction_interactionsource_id_fk FOREIGN KEY (interactionsource_id) REFERENCES ontologyterm (id);
+ALTER TABLE geneinteraction ADD CONSTRAINT geneinteraction_interactiontype_id_fk FOREIGN KEY (interactiontype_id) REFERENCES ontologyterm (id);
+ALTER TABLE geneinteraction ADD CONSTRAINT geneinteraction_interactorarole_id_fk FOREIGN KEY (interactorarole_id) REFERENCES ontologyterm (id);
+ALTER TABLE geneinteraction ADD CONSTRAINT geneinteraction_interactorbrole_id_fk FOREIGN KEY (interactorbrole_id) REFERENCES ontologyterm (id);
+ALTER TABLE geneinteraction ADD CONSTRAINT geneinteraction_interactoratype_id_fk FOREIGN KEY (interactoratype_id) REFERENCES ontologyterm (id);
+ALTER TABLE geneinteraction ADD CONSTRAINT geneinteraction_interactorbtype_id_fk FOREIGN KEY (interactorbtype_id) REFERENCES ontologyterm (id);
+CREATE INDEX geneinteraction_interactionsource_index ON geneinteraction USING btree (interactionsource_id);
+CREATE INDEX geneinteraction_interactiontype_index ON geneinteraction USING btree (interactiontype_id);
+CREATE INDEX geneinteraction_interactorarole_index ON geneinteraction USING btree (interactorarole_id);
+CREATE INDEX geneinteraction_interactorbrole_index ON geneinteraction USING btree (interactorbrole_id);
+CREATE INDEX geneinteraction_interactoratype_index ON geneinteraction USING btree (interactoratype_id);
+CREATE INDEX geneinteraction_interactorbtype_index ON geneinteraction USING btree (interactorbtype_id);
+ALTER TABLE geneinteraction ADD CONSTRAINT geneinteraction_interactionid_uk UNIQUE (interactionid);
+ALTER TABLE geneinteraction ADD CONSTRAINT geneinteraction_uniqueid_uk UNIQUE (uniqueid);
+
+CREATE TABLE genemolecularinteraction (
+	id bigint PRIMARY KEY,
+	aggregationdatabase_id bigint,
+	detectionmethod_id bigint
+);
+
+ALTER TABLE genemolecularinteraction ADD CONSTRAINT genemolecularinteraction_id_fk FOREIGN KEY (id) REFERENCES geneinteraction (id);
+ALTER TABLE genemolecularinteraction ADD CONSTRAINT genemolecularinteraction_aggregationdatabase_id_fk FOREIGN KEY (aggregationdatabase_id) REFERENCES ontologyterm (id);
+ALTER TABLE genemolecularinteraction ADD CONSTRAINT genemolecularinteraction_detectionmethod_id_fk FOREIGN KEY (detectionmethod_id) REFERENCES ontologyterm (id);
+CREATE INDEX genemolecularinteraction_aggregationdatabase_index ON genemolecularinteraction USING btree (aggregationdatabase_id);
+CREATE INDEX genemolecularinteraction_detectionmethod_index ON genemolecularinteraction USING btree (detectionmethod_id);
+
+CREATE TABLE geneinteraction_crossreference (
+	geneinteraction_id bigint,
```

Comment:
Does this table not get a `id bigint PRIMARY KEY,`?

Reply:
no, it's a join table

Reply:
but for instance isn't `genemolecularinteraction` a join table and it has an id?

Reply:
genemolecularinteraction isn't a join table

Reply:
GeneMolecularInteraction is the child class of GeneInteraction